cpu/cortexm_common/mpu.c

```diff
@@ -0,0 +1,72 @@
+/*
+ * Copyright (C) 2016 Loci Controls Inc.
+ *
+ * This file is subject to the terms and conditions of the GNU Lesser
+ * General Public License v2.1. See the file LICENSE in the top level
+ * directory for more details.
+ */
+
+/**
+ * @ingroup     cpu_cortexm_common
+ * @{
+ *
+ * @file        mpu.c
+ * @brief       Cortex-M Memory Protection Unit (MPU) Driver
+ *
+ * @author      Ian Martin <ian@locicontrols.com>
+ *
+ * @}
+ */
+
+#include <assert.h>
+
+#include "cpu.h"
+#include "mpu.h"
+
+int mpu_disable(void) {
+#if __MPU_PRESENT
+    MPU->CTRL &= ~MPU_CTRL_ENABLE_Msk;
+    return 0;
+#else
+    return 1;
+#endif
+}
+
+int mpu_enable(void) {
+#if __MPU_PRESENT
+    MPU->CTRL |= MPU_CTRL_PRIVDEFENA_Msk | MPU_CTRL_ENABLE_Msk;
+
+    /* Enable the memory fault exception */
+    SCB->SHCSR |= SCB_SHCSR_MEMFAULTENA_Msk;
+
+    return 0;
```

Comment:
This returns 0 on success, 1 on "not available", right?
I find the opposite more intuitive.